### ISSUE TITLE
Regression line assertion

### ISSUE THREAD

**Issue by nkorinek** [earthlab/matplotcheck#243]
Noticed a very rare issue with the regression line assertions. The assertion has this line to ensure that all x values are within the range of the expected line. 
```
if min(line_x_vals) <= min_val and max(line_x_vals) >= max_val:
```
Very rarely, matplotlib will be off in the wrong direction by .000000000000001, as seen here:
```
7.000000000000001 7.0 24.0 24.0
False True

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-10-123ae72b3d52> in <module>
     99 
    100 # Check line is correct
--> 101 line_figure_tests.assert_line(slope_exp=slope_data, intercept_exp=intercept_data)

~/GitHub/matplotcheck/matplotcheck/base.py in assert_line(self, slope_exp, intercept_exp, xtime, message_no_line, message_data)
   1106 
   1107         assert flag_exist, message_no_line
-> 1108         assert flag_length, message_data
   1109 
   1110     def assert_lines_of_type(self, line_types):

AssertionError: Line does not cover data set
```

A possible fix would be to use `math.isclose()` instead of the `<=` comparisons? Not sure. Very rare occurrence but common enough it's failing my lines vignette PR.

**Comment by nkorinek** [comment]
@lwasser new issue here, with your input I'll fix it! 

I would make a gist to demonstrate this, but I only found it because I was using `np.randint` to generate my plots, and I could only get the issue to come up by re-running the same code over and over until the numbers came up just wrong.

**Comment by nkorinek** [comment]
I made a gist to demonstrate. It seems to fail with a 0 in the x column everytime, but I don't think it's JUST 0. 0 does trigger it everytime however. https://gist.github.com/nkorinek/403614b560b6317c56a86c593a5dce71

**Comment by lwasser** [comment]
@nkorinek thank you - this definitely looks like a tolerance issue!! please go ahead and work on a fix!

**Comment by nkorinek** [comment]
Addressed by https://github.com/earthlab/matplotcheck/pull/248

**Comment by nkorinek** [comment]
This can be closed, fixed in #274 @lwasser

**Comment by lwasser** [comment]
closing!!
